src/api/Buffer.test.ts

```diff
@@ -69,159 +69,188 @@ describe('Buffer API', () => {
-      expect(await buffer.changedtick).toBe(initial + 2);
-    }));
+    it(
+      'gets changedtick of buffer',
+      withBuffer([], async buffer => {
+        const initial = await buffer.changedtick;
+
+        // insert a line
+        buffer.append('hi');
+        expect(await buffer.changedtick).toBe(initial + 1);
+
+        // clear buffer
+        buffer.remove(0, -1, false);
+        expect(await buffer.changedtick).toBe(initial + 2);
+      })
+    );
 
     it('gets the current buffer name', async () => {
-      const name = await (await nvim.buffers )[0].name;
+      const name = await (await nvim.buffers)[0].name;
       expect(name).toMatch('test.js');
     });
 
-    it('is a valid buffer', withBuffer([],async buffer => {
-      expect(await buffer.valid).toBe(true);
-    }));
-
-    it('sets current buffer name to "foo.js"', withBuffer( [],
-        async buffer => {
-            buffer.name = 'foo.js';
-            expect(await buffer.name).toMatch('foo.js');
-            buffer.name = 'test2.js';
-            expect(await buffer.name).toMatch('test2.js');
-    }));
-
-    it('can replace first line of buffer with a string', withBuffer(
-        ['foo'], async buffer => {
-            buffer.replace('test', 0);
-            expect(await buffer.lines).toEqual(['test']);
-    }));
-
-    it('can insert lines at beginning of buffer', withBuffer( ['test'],
-        async buffer => {
-            await buffer.insert(['test', 'foo'], 0);
-            expect(await buffer.lines).toEqual(['test', 'foo', 'test']);
-        })
+    it(
+      'is a valid buffer',
+      withBuffer([], async buffer => {
+        expect(await buffer.valid).toBe(true);
+      })
     );
 
-    it('can replace buffer starting at line 1', withBuffer(
-        ['test','b','c','d'],
-        async buffer => {
-            buffer.replace(['bar', 'bar', 'bar'], 1);
-            expect(await buffer.lines).toEqual(['test', 'bar', 'bar', 'bar']);
-        }
-    ));
-
-    it('inserts line at index 2', withBuffer(
-        ['test','bar','bar', 'bar'],
-        async buffer => {
-            buffer.insert(['foo'], 2);
-            expect(await buffer.lines).toEqual(['test', 'bar', 'foo', 'bar', 'bar']);
-        }
-    ));
-
-    it('removes last 2 lines', withBuffer(
-        ['test','bar','foo','a','b'],
-        async buffer => {
-            buffer.remove(-3, -1);
-            expect(await buffer.lines).toEqual(['test', 'bar', 'foo']);
-        },
-    ));
-
-    it('append lines to end of buffer', withBuffer(
-        [ 'test', 'bar', 'foo' ],
-        async buffer => {
-            await buffer.append(['test', 'test']);
-
-            expect(await buffer.lines).toEqual([
-                'test',
-                'bar',
-                'foo',
-                'test',
-                'test',
-            ]);
-    });
+    it(
+      'sets current buffer name to "foo.js"',
+      withBuffer([], async function buffer() {
+        buffer._name = 'foo.js';
```

Comment:
Changed callback to named `buffer` function because buffer in `withBuffer` is self-contained. Didn't try `nvim.buffer`. `name` is a reserved word for functions so I replaced it with `_name`. This change applies to other tests in this file.